### PR TITLE
[ML] Delete unused data frame analytics state

### PR DESCRIPTION
**Pull request by dimitris-athanasiou**
This commit adds removal of unused data frame analytics state
from the _delete_expired_data API (and in extend th ML daily
maintenance task). At the moment the potential state docs
include the progress document and state for regression and
classification analyses.

### CODE REVIEW

File: x-pack/plugin/ml/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/ml/integration/ClassificationIT.java [hunk 1]
```diff
@@ -315,6 +318,36 @@ public void testTwoJobsWithSameRandomizeSeedUseSameTrainingSet() throws Exceptio
...
+        startAnalytics(jobId);
+        waitUntilAnalyticsIsStopped(jobId);
+
+        assertProgress(jobId, 100, 100, 100, 100);
+        assertThat(searchStoredProgress(jobId).getHits().getTotalHits().value, equalTo(1L));
+        assertModelStatePersisted(stateDocId());
+        assertInferenceModelPersisted(jobId);
+
+        // Delete the config straight from the config index
+        DeleteResponse deleteResponse = client().prepareDelete(".ml-config", DataFrameAnalyticsConfig.documentId(jobId))
```
Comment (davidkyle) [inline x-pack/plugin/ml/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/ml/integration/ClassificationIT.java]: Why delete the config? Does `DeleteExpiredDataAction` require the config to be deleted

Comment (dimitris-athanasiou) [reply x-pack/plugin/ml/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/ml/integration/ClassificationIT.java]: Yes, for the state to become "unused", ie. there is no job owning those state docs.

**Review by davidkyle — APPROVED** [review]
LGTM

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/analyses/Classification.java [hunk 2]
```diff
@@ -39,6 +39,8 @@
     public static final ParseField TRAINING_PERCENT = new ParseField("training_percent");
     public static final ParseField RANDOMIZE_SEED = new ParseField("randomize_seed");
 
+    private static final String STATE_DOC_ID_SUFFIX = "_classification_state#1";
```
[the remark this answers is not in the document]
Comment (dimitris-athanasiou) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/analyses/Classification.java]: I also thought about this. We currently handle just a single doc. But we can fix this if and when we switch to handling state split over multiple docs.

**Review by davidkyle — APPROVED** [review]
LGTM

**Pull request merged**